sound/soc/soc-pcm.c

```diff
@@ -3176,6 +3184,14 @@ int soc_new_pcm(struct snd_soc_pcm_runtime *rtd, int num)
 		rtd->ops.ioctl		= soc_pcm_ioctl;
 	}
 
+	if (rtd->dai_link->no_pcm) {
+		if (playback)
+			pcm->streams[SNDRV_PCM_STREAM_PLAYBACK].substream->private_data = rtd;
+		if (capture)
+			pcm->streams[SNDRV_PCM_STREAM_CAPTURE].substream->private_data = rtd;
+		goto out;
+	}
+
```

Comment:
@lyakh need a comment here for this change. Why add pcm ops for a dai_link with no_pcm?

Reply:
I need the .prepare() hook to configure the hardware for the VM operation. I call it when the TRIGGER -> START IPC is executed. I'll add a comment